apis/controlplane/v1alpha3/nestedcontrolplanetemplate_types.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1alpha1 "sigs.k8s.io/cluster-addons/installer/pkg/apis/config/v1alpha1"
+)
+
+// NestedControlPlaneTemplateSpec defines the desired state of NestedControlPlaneTemplate
+type NestedControlPlaneTemplateSpec struct {
+	// KubernetesVersion allows you to specify the Kubernetes version this
+	// template will provision
+	KubernetesVersion KubernetesVersion `json:"version"`
+
+	// Components allows you to specify the multiple components necessary to
+	// provision a Virtual Control Plane
+	// +optional
+	Components []configv1alpha1.Addon `json:"components,omitempty"`
+
+	// Default allows you to set this NestedTemplate as a a default template,
+	// by setting this to true any new NestedClusters will be created without
+	// a TemplateRef will be deployed with this template.
+	Default bool `json:"default,omitempty"`
```

Comment:
I'd expect something like default templates to be a feature we add further down the line - initially, in this early development phase, requiring users to specify something explicitly works fine. If we introduce this without consideration, we end out not answering questions like "what if multiple templates exist that are all marked default?"

Reply:
Use a validating webhook to prevent multiple templates from being marked as default?